x-pack/solutions/observability/plugins/streams/server/lib/streams/data_streams/manage_data_streams.ts

```diff
@@ -129,6 +129,10 @@ export async function updateDataStreamsLifecycle({
     const dataStreams = await esClient.indices.getDataStream({ name: names });
     const isIlm = isIlmLifecycle(lifecycle);
 
+    await Promise.all(
+      names.map((name) => retryTransientEsErrors(() => esClient.ilm.removePolicy({ index: name })))
+    );
```

Comment:
Deleting any effective policy ensures a clean state in case we transition to a new policy. By doing so we make no index gets in an undesired state with the drawback that all indices will go through all the new phases starting with hot.

We can potentially avoid that by diffing the previous and new policies, but I'll leave that to a separate issue